File: ALT/Lp2 cpp/SelectionSort.cpp
#include <bits/stdc++.h>
using namespace std;
void printArray(vector<int>&arr, int size)
{
    int i;
    for (i = 0; i < size; i++)
    {
        cout << arr[i] << " ";
    }
    cout<<endl;
}
// Function for Selection sort
void selectionSort(vector<int>&arr, int n)
{
    int i, j, min_idx;
    for (i = 0; i < n - 1; i++)
    {
        min_idx = i;
        for (j = i + 1; j < n; j++)
        {
            if (arr[j] < arr[min_idx])
                min_idx = j;
        }
        if (min_idx != i)
            swap(arr[min_idx], arr[i]);
        cout<<"After iteration"<<i+1<<": ";
        printArray(arr,n);
    }
}

int main()
{
    int n;

    cout<<"enter size of array: ";
    cin>>n;
    vector<int> arr(n);
    // Function Call
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    selectionSort(arr, n);
    cout << "Sorted array: \n";
    printArray(arr, n);
    return 0;
}

// This is code is contributed by rathbhupendra

File: N/Selectionsort.java

import java.util.Arrays;
public class Selectionsort{
    public static void selectionsort(int arr[]){
        for(int i=0; i<arr.length; i++){
            int min=i;
            for(int j=i+1; j<arr.length-1; j++){
                if(arr[j]<arr[min]){
                    min = j;
                }
            }
            //swap
            int temp = arr[min];   
            arr[min] = arr[i];  
            arr[i] = temp;  
        }
    }
    public static void main(String args[]){
        int arr[] = {12, 25, 12, 22, 64};
        selectionsort(arr);
        System.out.println("Sorted array:" +  Arrays.toString(arr));
       


    }
}